akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala

```diff
@@ -152,19 +152,13 @@ private[remote] class ArteryTcpTransport(_system: ExtendedActorSystem, _provider
         }
       }
 
-      if (streamId == ControlStreamId) {
-        // restart of inner connection part important in control flow, since system messages
-        // are buffered and resent from the outer SystemMessageDelivery stage.
-        RestartFlow.withBackoff[ByteString, ByteString](
-          settings.Advanced.OutboundRestartBackoff,
-          settings.Advanced.GiveUpSystemMessageAfter, 0.1)(flowFactory)
-      } else {
-        // Best effort retry a few times
-        RestartFlow.withBackoff[ByteString, ByteString](
-          settings.Advanced.OutboundRestartBackoff,
-          settings.Advanced.OutboundRestartBackoff * 5, 0.1, maxRestarts = 3)(flowFactory)
-      }
-
+      val maxRestarts = if (streamId == ControlStreamId) Int.MaxValue else 3
```

Comment:
If we gave up the messages after "settings.Advanced.GiveUpSystemMessageAfter", why we retry infinitely to re-establish the control association? Or you are giving up in some other part of the code?

Reply:
GiveUpSystemMessageAfter is used by the SystemMessageDelivery stage to resend system messages like Watch and DeathWatchNotifications. Using it here was wrong.

Note that the `maxBackoff` parameter to `RestartFlow.withBackoff` is not a give up, it's the upper bound of the backoff interval.